runtime/src/iree/base/internal/wait_handle_promise.c

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 The IREE Authors
```

Comment:
I think we'll want to rename this to wait_handle_emscripten - promises are one kind of handle we'll support on the web but we will likely want others too (futex, specifically) and "promise" doesn't really match that this is for "the emscripten platform"

Reply:
hmmm maybe JAVASCRIPT_PROMISE? naming is hard :) trying to think of a future where we don't use emscripten (so the emscripten-specific wait handle/loop/etc may have alternatives but they would still use the same promise wait handle enum)